app/models/katello/content_view_version.rb

```diff
@@ -311,11 +311,13 @@ def package_groups
 
     def update_content_counts!
       self.content_counts = {}
-      RepositoryTypeManager.indexable_content_types.map(&:model_class).each do |content_type|
-        if content_type::CONTENT_TYPE == DockerTag::CONTENT_TYPE
+      RepositoryTypeManager.indexable_content_types.each do |content_type|
+        if content_type&.model_class::CONTENT_TYPE == DockerTag::CONTENT_TYPE
           content_counts[DockerTag::CONTENT_TYPE] = docker_tags.count
+        elsif content_type&.model_class::CONTENT_TYPE == GenericContentUnit::CONTENT_TYPE
+          content_counts[content_type.content_type] = content_type&.model_class.in_repositories(self.repositories.archived).select { |unit| unit.content_type == content_type.content_type }.count
```

Comment:
.select{} would retrieve all the content types and then iterate over each one which works but would be slow.  A better way would be to use the query structure:
```
.where(:content_type => content_type.content_type).count
```

that executes it in the database via sql and should be much faster